### PR TITLE
remove dead code & replace remaining parse_ident

### PR DESCRIPTION
**Pull request by burrbull**
last commit from #177

### CODE REVIEW

**Comment by crlf0710** [comment]
Seems good to me, but since there's quite a few pointer operation stuff, it'll be best to have more eyes on this.

File: dpx/src/specials/html.rs [hunk 1]
```diff
@@ -737,76 +720,60 @@ unsafe fn spc_handler_html_default(mut spe: *mut spc_env, mut ap: *mut spc_arg)
...
+    while p[0] != 0 && libc::isspace(p[0] as _) != 0 {
+        p = &p[1..];
     }
-    /* parsed transformation key */
-    let mut k = 0;
-    while !_TKEYS[k].is_null() && strcmp(q, _TKEYS[k]) != 0 {
-        k += 1
+    let q = p.parse_c_ident();
+    if q.is_none() {
+        return Err(());
```
Comment (cormacrelf) [inline dpx/src/specials/html.rs:725]: `?` operator works too

File: dpx/src/specials/html.rs [hunk 2]
```diff
@@ -737,76 +720,60 @@ unsafe fn spc_handler_html_default(mut spe: *mut spc_env, mut ap: *mut spc_arg)
     error
 }
 /* translate wsp* '(' wsp* number (comma-wsp number)? wsp* ')' */
```
Comment (cormacrelf) [inline dpx/src/specials/html.rs:713]: Another piece of code where `nom` might make things a lot cleaner. The parse traits would be usable, with some small wrappers that clone the slice pointer, mutate it, subtract the length from the input and return the remainder and output as a `nom::IResult` by converting to a basic nom error. Then functions like these would be a few lines long, using nom combinators (except the end bit that mutates, just return the data and use it to mutate elsewhere).

**Review by cormacrelf — COMMENTED** [review]
Pointer stuff looks fine

**Comment by crlf0710** [comment]
Need a rebase, i think...

**Comment by burrbull** [comment]
Done.

**Comment by crlf0710** [comment]
Merging~ Thanks~

**Pull request merged**